src/s11/handlers/ddn_ack_handler.c

```diff
@@ -46,14 +45,19 @@ struct MsgBuffer* ddnAckMsgBuf_p = NULL;
 static int
 ddn_ack_processing(struct DDN_ACK_Q_msg *ddn_ack_msg)
 {
+	struct MsgBuffer* ddnAckMsgBuf_p = createMsgBuffer(1024);
+	if(ddnAckMsgBuf_p == NULL)
+	{
+	    log_msg(LOG_ERROR, "Error in initializing msg buffers required by gtp codec.\n");
+            return -1;
+	}
 	GtpV2MessageHeader gtpHeader;
 	gtpHeader.msgType =  GTP_DOWNLINK_DATA_NOTIFICATION_ACK;
 	gtpHeader.sequenceNumber = ddn_ack_msg->seq_no;
 	gtpHeader.teidPresent = true;
-	gtpHeader.teid = ddn_ack_msg->s11_sgw_c_fteid.header.teid_gre;
+	gtpHeader.teid = ddn_ack_msg->s11_sgw_cp_teid;
     struct sockaddr_in sgw_ip = {0};
-    create_sock_addr(&sgw_ip, g_s11_cfg.egtp_def_port,
-                    ddn_ack_msg->s11_sgw_c_fteid.ip.ipv4.s_addr);
+    create_sock_addr(&sgw_ip, g_s11_cfg.egtp_def_port, htonl(g_s11_cp_addr.sin_addr.s_addr));
```

Comment:
using g_s11_cp_addr causes issues when Mme talks to multiple spgw. If we can store the ip from which we received the ddn message, we can send back on the same.

Reply:
Taken care in https://github.com/omec-project/Nucleus/pull/54/commits/f72e5b13291453dbd0ab067828c41aaa03d80e62